scipy/interpolate/_bsplines.py

```diff
@@ -561,6 +561,87 @@ def integrate(self, a, b, extrapolate=None):
         integral *= sign
         return integral.reshape(ca.shape[1:])
 
+    def design_matrix(self, x, kind="dense", check_finite=True):
```

Comment:
Also, `check_finite`. Is it really needed. The knots cannot have a inf/nan value -- this should raise ; the `x` values can have a nan --- then the whole row is nans but this (i) seems appropriate (there is no way we can assign the "missing" semantics to a nan, so let's not even try), and (ii) does not seem to be able to cause a crash. 

So maybe just keep it simple and let all nans through. If really needed, this can be bolted on later.